tutorial.md

```diff
@@ -0,0 +1,156 @@
+# Create a new example
+
+We will walk through the steps for a new example Jupyter Notebook
+
+## Create a notebook
+
+Continue in the directory where you cloned the repository.
+Create a Jupyter notebook in an existing folder.
+
+### Start Jupyter
+
+Change to a subdirectory and start the Jupyter notebook server:
+```
+cd idaes_examples/nb/flowsheets
+jupyter notebook &
+```
+
+### Add a markdown cell
+
+Then, in the Jupyter notebook browser, create a new blank note book ("New" -> "IPython 3 kernel").
+Name the notebook `my_notebook_src` and make the first cell a Markdown cell with the following text:
+```
+# My notebook
+This is my example notebook.
+```
+
+### Add code cells
+
+Add 3 code cells with the following content (one statement per cell):
+```
+import idaes
+
+print(idaes.__version__)
+
+assert idaes.__version__.startswith('2')
+```
+
+### Add tags
+
+To demonstrate how the tags work, we will add tags to a couple of these cells:
+1. To see current tags, select "View" -> "Cell toolbar" -> "Tags".
+2. Add the tag `noauto` to the cell with the print statement
+3. Add the tag `testing` to the cell with the assert statement
+
+### Save work
+
+Finally, save the notebook.
+
+## Add the example to the table of contents
+
+For building and testing to 'see' a notebook it needs to be in the Jupyterbook table of contents.
+The following steps add the new notebook to the table of contents.
+
+1. Open `idaes_examples/nb/_toc.yml` in an editor.
+2. Find `- file: flowsheets/index` under `chapters:`.
+3. In the list of sections for that chapter, add `- file: flowsheets/my_notebook_doc` (indent the same as the other sections).
+4. Save the modified file.
+
+Note: Although the notebook was called `my_notebook_src`, the filename in the table of contents entry is `my_notebook_doc`.
+
+
+## Test the example
+
+Example notebooks are tested with `pytest`, using a special configuration that is stored in the root directory of the repository (in the file "pyproject.toml").
+
+Change to the top directory for the repository, then test your new example notebook with: 
+```
+pytest -k my_notebook
+```
+
+Why `-k`? Although you can test *all* notebooks with the simple command `pytest`, since notebooks are found and tested with an extra level of indirection you can't simply name the notebook directly. Instead, testing a single notebook requires using a "-k" expression that will match the notebook.
+
+## Run 'unit' tests for all examples
+
+To do quick sanity checks and other tests of the example notebooks, there are standard Python test files under `idaes_examples/nb`. To run those *instead* of the example notebooks, simply run pytest from the `idaes_examples` sub-directory:
+```
+cd idaes_examples
+pytest
+```
+
+Note: This works because there is a "pytest.ini" in that directory that is used instead of the configuration in the root of the repository. That configuration does not set up the special ('nbmake') configuration for running Jupyter notebooks under pytest.
```

Comment:
Depending on how much we manage to streamline the testing configuration, this step could be simplified (i.e. running `pytest --static-checks`, or `pytest --skip-execution` or similar).

Reply:
if there's a nicer way to do this, I'm fine with it.